### ISSUE TITLE
Using == operator in symbolic expression tests

### ISSUE THREAD

**Issue by sylee957** [sympy/sympy#16089]
```
from sympy import *

n = Symbol('n')

lhs = -3/(n**2 + 18*n)
rhs = -3/(n*(n + 18))
lhs == rhs
```
It gives `False`.
The equals operator in `Basic` objects seems like compares hash values, rather than having any mathematical validity.
And it seems like this is actually a intended design.
https://github.com/sympy/sympy/wiki/Faq#why-does-sympy-say-that-two-equal-expressions-are-unequal

However, a lot of tests in `solvers/tests/test_solver.py` are using this comparison to test out expressions, and they may give failings whenever the solver routine gets modified.
Would it be better to replace them to relational `Eq()`, and use `simplify` to compare them in more canonical way?

**Comment by bharatnishant** [comment]
@asmeurer Kindly review this request. If it needs working then I would like to work on it

**Comment by asmeurer** [comment]
This requires more discussion before any work is to be done.

**Comment by oscarbenjamin** [comment]
The same thing happens in `test_ode.py`. I think it's good (most of the time) that the tests also test the form of the solution returned. It means that you can see how a PR changes the output returned from `dsolve`/`solve` etc.

Ideally we should avoid changing the form of the returned solution unless it is an improvement. Sometimes it's not possible to fix one issue without changing the form of the returned solution in other cases. The fact that the tests have to be fixed before the PR is merged allows reviewers to see what is changed - human judgment is needed to decide whether the change is acceptable.

**Comment by asmeurer** [comment]
I agree. Furthermore, simplify() can be an expensive function and it would not be good to call it in the tests.
